packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -3047,12 +3067,49 @@ export function captureCommitPhaseError(sourceFiber: Fiber, error: mixed) {
           markRootUpdated(root, SyncLane, eventTime);
           ensureRootIsScheduled(root, eventTime);
           schedulePendingInteractions(root, SyncLane);
+        } else {
+          // This component has already been unmounted.
+          // We can't schedule any follow up work for the root because the fiber is already unmounted,
+          // but we can still call the log-only boundary so the error isn't swallowed.
+          // TODO We will need to rethink this for function component boundaries
+          if (
+            typeof instance.componentDidCatch === 'function' &&
+            !isAlreadyFailedLegacyErrorBoundary(instance)
+          ) {
+            try {
+              instance.componentDidCatch(error, errorInfo);
+            } catch {}
+          }
         }
         return;
       }
     }
     fiber = fiber.return;
   }
+
+  if (!foundErrorBoundary) {
+    if (nearestMountedAncestor !== null) {
+      // An error was thrown inside of an unmounted subtree that contained no error boundaries.
+      // Bubble up to the nearest still-mounted error boundary for loggin purposes.
+      fiber = nearestMountedAncestor;
+      while (fiber !== null) {
+        if (fiber.tag === ClassComponent) {
+          const instance = fiber.stateNode;
+          if (
+            typeof instance.componentDidCatch === 'function' &&
+            !isAlreadyFailedLegacyErrorBoundary(instance)
+          ) {
+            const errorInfo = createCapturedValue(error, sourceFiber);
+            try {
+              instance.componentDidCatch(error, errorInfo);
```

Comment:
Should we do more here? should we actually call gDSFE and schedule a follow-up render? We *could* but it's not clear to me that it's the *right* thing to do.